File: Lecture04/ListViews/app/src/main/java/com/codingblocks/listviews/MainActivity.java
package com.codingblocks.listviews;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.widget.ArrayAdapter;
import android.widget.ListView;

public class MainActivity extends AppCompatActivity {

    String[] courses = new String[] {
            "Launchpad", "Crux", "Webdev NodeJS",
            "Android", "Django", "Machine Learning",
            "Competitive Coding", "Java for Beginners",
            "C++ for Beginners", "Algo++", "Algo.Java",
            "Web Development with Java", "Angular", "React",
            "Flutter", "iOS"
    };
    ListView lvCourses;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        lvCourses = findViewById(R.id.lvCourses);
        ArrayAdapter<String> courseAdapter = new ArrayAdapter<>(
                this,
                android.R.layout.simple_list_item_1,
                android.R.id.text1,
                courses
        );
        lvCourses.setAdapter(courseAdapter);
    }
}

File: Lecture05/KotlinRecyclerView/app/src/main/java/com/codingblocks/kotlinrecyclerview/MainActivity.kt
package com.codingblocks.kotlinrecyclerview

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.support.v7.widget.GridLayoutManager
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        rvCourses.layoutManager = GridLayoutManager(
                this, 5,
                RecyclerView.HORIZONTAL,
                false
        )
        rvCourses.adapter = CourseAdapter(genRandomCourses(200))
    }
}
